integration/integration_test.go

```diff
@@ -279,6 +279,14 @@ EOF
 				. testenv1/bin/activate-hermit
 				assert test "$(./testenv2/bin/testbin1)" = "testbin1 1.0.1"
 			`},
+		{name: "ExecuteFromOtherEnvironmentLoadsDependentEnvars",
+			preparations: prep{allFixtures("testenv1", "testenv2")},
+			script: `
+				testenv2/bin/hermit install testbin1-1.0.0 testbin4
+				. testenv1/bin/activate-hermit
+				hermit install testbin1-1.0.1
+				assert test "$(./testenv2/bin/testbin4)" = "env[1.0.0] exec[testbin1 1.0.0]"
+			`},
```

Comment:
I would like to see a test setup similar to this, but in a nested environment (like the `environment_inheritance` test data) as this is the scenario we saw failing (nested environment calling parent binary with a call to an overwritten dependency).